pkg/core/provider.go

```diff
@@ -85,10 +94,38 @@ func NewProviderFromArchive(filename string) (Provider, error) {
 	}, nil
 }
 
+// SigningKeys represents the signing-keys.json that we expect in the storage backend
+// https://github.com/TierMobility/boring-registry#gpg-public-key-format
 type SigningKeys struct {
 	GPGPublicKeys []GPGPublicKey `json:"gpg_public_keys,omitempty"`
 }
 
+// IsValidSha256Sums verifies whether the GPG signature of to the SHA256SUMS file was created with a private key
+// corresponding to one of the public keys in SigningKeys
+func (s *SigningKeys) IsValidSha256Sums(sha256Sums, sha256SumsSig []byte) error {
+	for _, key := range s.GPGPublicKeys {
+		keyring, err := openpgp.ReadArmoredKeyRing(strings.NewReader(key.ASCIIArmor))
+		if err != nil {
+			return fmt.Errorf("error reading signing key: %w", err)
+		}
+
+		_, err = openpgp.CheckDetachedSignature(keyring, bytes.NewReader(sha256Sums), bytes.NewReader(sha256SumsSig), nil)
+
+		// If the signature issuer does not match the key, keep trying the rest of the provided keys.
+		if errors.Is(err, openpgpErrors.ErrUnknownIssuer) {
```

Comment:
Do we want to surface this error or log this as an error?

Reply:
If it's an `openpgpErrors.ErrUnknownIssuer`, we do not want to return it, as it's an expected error :)